lib/ride/services/audio.dart

```diff
@@ -0,0 +1,820 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:audio_session/audio_session.dart';
+import 'package:collection/collection.dart';
+import 'package:flutter_tts/flutter_tts.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:priobike/main.dart';
+import 'package:priobike/positioning/services/positioning.dart';
+import 'package:priobike/ride/interfaces/prediction.dart';
+import 'package:priobike/ride/messages/prediction.dart';
+import 'package:priobike/ride/models/audio.dart';
+import 'package:priobike/ride/models/recommendation.dart';
+import 'package:priobike/ride/services/ride.dart';
+import 'package:priobike/routing/models/instruction.dart';
+import 'package:priobike/routing/models/route.dart';
+import 'package:priobike/settings/models/speed.dart';
+import 'package:priobike/settings/services/settings.dart';
+import 'package:priobike/tracking/models/speed_advisory_instruction.dart';
+import 'package:priobike/tracking/services/tracking.dart';
+
+const String redInText = "rot in";
+const String greenInText = "grün in";
+
+/// The distances before the crossings when a speed advisory instruction should be played.
+const List<int> speedAdvisoryDistances = [300, 100];
+
+/// The distances until an instruction is played even if the distance is smaller than the speedAdvisoryDistances.
+const List<int> speedAdvisoryMinDistances = [200, 50];
+
+/// The minimal distance before the crossing when a speed advisory instruction should be played. (Except wait for green)
+const int speedAdvisoryMinDistance = 50;
```

Comment:
Based on the given documentations I'm still not 100% sure what those thresholds are doing. Could you explain them to me and maybe we can think of better docs together?